rmsutil/crypto.py

```diff
@@ -0,0 +1,42 @@
+import binascii
+import hashlib
+import hmac
+import os
+
+import six
+
+
+def get_rand_hex(entropy_bytes):
+    """Returns a random hexadecimal string.
+
+    Args:
+        entropy_bytes (str): Number of bytes of entropy to encode.
+
+    Returns:
+        str: A random hex string, twice the length of entropy_bytes.
+    """
+    return binascii.hexlify(os.urandom(entropy_bytes))
+
+
+def hmac_sha256_128(key, message):
+    """Returns an HMAC-SHA256 given a 128-bit key.
+
+    Args:
+        key (str): 128-bit signing key encoded in hexadecimal.
+        message (str or bytes): Arbitrary message to hash. If a Unicode
+            string, will be encoded as UTF-8 before hashing.
+
+    Returns:
+        str: Hexadecimal-encoded HMAC of the message
+    """
+
+    if len(key) != 32:  # 128 / 8 * 2
```

Comment:
Why does this require it to be hex encoded? Also, why does it require the key be 128-bit? Presumably to avoid the hashing step if the key is too long, but the "block" size of SHA256 is 512-bit so this restriction makes little sense. Maybe the restriction should be not less than 128-bit?